### PR TITLE
0.2.X deprecation

### PR DESCRIPTION
**Pull request by nshaaban-cPacket**
This will be the final update to 0.2.X before 0.3.0 goes live. Uses deprecation notes to walk the user through how to upgrade from 0.2.X to 0.3.0. AKA if the users resolve all deprecation warnings described in the notes they can upgrade to 0.3.0 without errors. (even without looking at the changelog)

### CODE REVIEW

File: src/lib.rs [hunk 1]
```diff
@@ -177,6 +177,7 @@ impl SocketBuilder {
 
     /// Sets the target namespace of the client. The namespace must start
     /// with a leading `/`. Valid examples are e.g. `/admin`, `/foo`.
+    #[deprecated(note = "use namespace instead")]
```
Comment (1c3t3a) [inline src/lib.rs:180]: We might write 'use `namespace(nsp)` instead' in order to indicate that this refers to a function.

File: src/lib.rs [hunk 2]
```diff
@@ -239,11 +247,37 @@ impl SocketBuilder {
     ///     .connect();
     ///
     /// ```
+    #[deprecated(note = "use tls_config instead")]
```
Comment (1c3t3a) [inline src/lib.rs:250]: Same like the nsp thing above.

File: src/lib.rs [hunk 3]
```diff
@@ -261,6 +295,7 @@ impl SocketBuilder {
     ///     .connect();
     ///
     /// ```
+    #[deprecated(note = "use opening_header instead")]
```
Comment (1c3t3a) [inline src/lib.rs:298]: Same as above

**Review by 1c3t3a — COMMENTED** [review]
This generally looks good with just a few points. Do you think it makes sense to add a `since = "0.3.0"` to the deprecation macros?

File: src/lib.rs (hunk 1 above)
Comment (nshaaban-cPacket) [reply src/lib.rs:180]:
```suggestion
     #[deprecated(note = "use namespace(nsp) instead")]
```

File: src/lib.rs (hunk 3 above)
Comment (nshaaban-cPacket) [reply src/lib.rs:298]:
```suggestion
     #[deprecated(note = "use opening_header(header, value) instead")]
```

File: src/lib.rs (hunk 2 above)
Comment (nshaaban-cPacket) [reply src/lib.rs:250]:
```suggestion
     #[deprecated(note = "use tls_config(tls_config) instead")]
```

**Review by 1c3t3a — APPROVED** [review]
Alright, looks good!

**Pull request merged**